apps/condo/domains/common/hooks/useSearch.ts

```diff
@@ -0,0 +1,28 @@
+import { useCallback, useState } from 'react'
+import { useRouter } from 'next/router'
+import qs from 'qs'
+import { pickBy, get, debounce } from 'lodash'
+import { getFiltersFromQuery } from '@condo/domains/common/utils/helpers'
+
+export const useSearch = (loading): [string, (search: string) => void] => {
+    const router = useRouter()
+    const filtersFromQuery = getFiltersFromQuery(router.query)
```

Comment:
And add specific IFilters generic arguments at the places where `getFiltersFromQuery` is being used.

`getFiltersFromQuery<IFilters>`
`useSearch<IFilters>`